docs/src/main/paradox/index.md

```diff
@@ -1,273 +1,93 @@
 
-You can start more cluster frontend nodes using port numbers between 3000-3999:
+## What happens when you run it
 
-```
-sbt "runMain worker.Main 3002"		
-```
+When `Main` is run without any parameters, it starts 6 `ActorSystem`s in the same JVM which then form a single cluster. 
 
-Note that this sample runs the 
-<a href="http://doc.akka.io/docs/akka/2.4.0/scala/persistence.html#Shared_LevelDB_journal" target="_blank">shared LevelDB journal</a>
-on the node with port 2551. This is a single point of failure, and should not be used in production. 
-A real system would use a 
-<a href="http://akka.io/community/" target="_blank">distributed journal</a>.
+Two of the nodes have the role `frontend` and simulate having an external interface, such as a REST API, where workloads can be posted. 
 
-The files of the shared journal are saved in the target directory and when you restart
-the application the state is recovered. You can clean the state with:
-
-```bash
-sbt clean
-```
+Two of the nodes has the role `worker` and starts two worker actors each that accept and process workloads.
 
-## Many Masters
+Two nodes have the role `backend` and contain an actor called `Master` which coordinates workloads from the frontend nodes, keep track of the workers, and delegate work to available workers. If one of these nodes goes down, the other one takes over its responsibilities.
 
-If the singleton master becomes a bottleneck we can start several master actors and
-shard the jobs among them. For each shard of master/standby nodes we use a separate
-cluster role name, e.g. "backend-shard1", "backend-shard2".
-Implementation of this is left as an exercise.
+A bird's eye perspective of the architecture would look like this:
 
-## Next Steps
+![Overview](images/cluster-nodes.png)
 
-In this example we have used
-<a href="http://doc.akka.io/docs/akka/2.4.0/scala/cluster-singleton.html"
-target="_blank">Cluster Singleton</a>,
-<a href="http://doc.akka.io/docs/akka/2.4.0/scala/cluster-client.html"
-target="_blank">Cluster Client</a> and
-<a href="http://doc.akka.io/docs/akka/2.4.0/scala/distributed-pub-sub.html"
-target="_blank">Distributed Publish Subscribe</a>.
-</p>
+The application was designed like this to support the following requirements:
 
-More in depth documentation can be found in the
-<a href="http://doc.akka.io/docs/akka/2.4.0/common/cluster.html"
-target="_blank">Cluster Specification</a> and in the
-<a href="http://doc.akka.io/docs/akka/2.4.0/scala/cluster-usage.html"
-target="_blank">Cluster Usage</a> documentation.
\ No newline at end of file
+ * elastic addition/removal of frontend nodes that receives work from clients
+ * elastic addition/removal of worker actors and worker nodes
+ * each worker hosting one or more workers
+ * the number of workers on each worker should not have to be the same across all workers 
+ * jobs should not be lost, and if a worker fails, the job should be retried
+  
+The design is based on Derek Wyatt's blog post 
```

Comment:
Probably since you want to credit Derek, it would be better to put this note in the intro?